lib/AST/Decl.cpp

```diff
@@ -2090,6 +2090,14 @@ bool swift::conflicting(ASTContext &ctx,
   if (!conflicting(sig1, sig2, skipProtocolExtensionCheck))
     return false;
 
+  // Static functions do not conflict with enum element decls if the types
+  // are different
+  if (sig1.IsStaticFunction == sig2.IsEnumElement) {
```

Comment:
Can we fold this into the following 'if'? I think even non-static functions should not conflict with enum elements.

In fact, is there any harm to treating enum elements as _functions_ for purposes of disambiguation?

Reply:
I agree, I think we should treat functions and enum elements the same for the purposes of overload signature checking. How about renaming `IsFunction` to something like `IsFunctionLike`, and then initialising it with `isa<AbstractFunctionDecl>(this) || isa<EnumElementDecl>(this)`?